trlx/trainer/accelerate_base_trainer.py

```diff
@@ -542,21 +542,21 @@ def learn(self):  # noqa: C901
 
         # For each epoch
         for _ in range(self.config.train.epochs):
-            # For each batch
-            for mbs in MiniBatchIterator(self.train_dataloader, self.mb_size, self.num_mb):
-                # For each update per batch
-                for _ in range(self.n_updates_per_batch):
-                    # Note that whereas standard policy gradient methods perform one
-                    # gradient update per batch, PPO for example commonly performs
-                    # multiple gradient updates on the same batch of data.
-                    # https://arxiv.org/pdf/1707.06347.pdf
-                    forward_time = 0
-                    backward_time = 0
+            # For each ppo epoch
+            for _ in range(self.n_inner_epochs):
+                # Note that whereas standard policy gradient methods perform one
+                # gradient update per batch, PPO for example commonly performs
+                # multiple epochs of gradient updates on the same batch of data.
+                # https://arxiv.org/pdf/1707.06347.pdf
+                # For each batch
+                for minibatch in MiniBatchIterator(list(self.train_dataloader), self.mb_size, self.num_mb):
```

Comment:
Nitpick: is there a reason it has to be changed to `list(self.train_dataloader)`?

Reply:
Of course, that makes sense! Also dumb question: will `shuffle=True` in `self.train_dataloader` produce optimal behaviour or is it different?

Reply:
I've added a commit that abstracts the training dataloader creation and then reinstantiates it at every ppo epoch. It shoudn't have changed the logic for any of the trainers apart from the accelerate_ppo_trainer.